src/throttler/throttler.ts

```diff
@@ -91,20 +96,39 @@ export abstract class Throttler<T> {
     return p;
   }
 
-  public add(task: T): void {
+  /**
+   * Add the task to the throttler.
+   * When the task is completed, resolve will be called with handler's result.
+   * If this task fails after retries, reject will be called with the error.
+   */
+  public add(
```

Comment:
FWIW, `public` is the default: https://www.typescriptlang.org/docs/handbook/classes.html

I'm thinking we should drop the `public` keyword - what do you think? We don't have a lint rule enforcing this... yet...

Reply:
I think that makes sense too. Let me take a look and I'll get back to you.